### PR TITLE
(feat) configurable versions

### PR DESCRIPTION
**Pull request by kirwea**
# Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [ ] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary

OHRI-1918
Added the version number and partner ohri logo

## Screenshots
Required if you are making UI changes. -->
Screenshot from 2023-12-20 11-27-03

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "packages/tooling/*"
   ],
   "scripts": {
-    "run:omrs": "openmrs",
+    "run:omrs": "openmrs develop --backend https://ohri-namibia-dev.globalhealthapp.net --sources packages/apps/esm-login-app/",
```
Comment (pirupius) [inline package.json:12]: Please revert this before making the PR to core

File: packages/apps/esm-login-app/__mocks__/config.mock.ts [hunk 2]
```diff
@@ -20,4 +20,19 @@ export const mockConfig: ConfigSchema = {
...
+  showVersionNumber: {
+    type: false,
+    _default: false,
+    _description: '',
+  },
+  partnerLogo: {
+    type: '',
+    _default: [],
+    _description: '',
+  },
```
Comment (pirupius) [inline packages/apps/esm-login-app/__mocks__/config.mock.ts:37]: Update these to include test data

File: packages/apps/esm-login-app/src/config-schema.ts [hunk 3]
```diff
@@ -1,6 +1,22 @@
 import { validators, Type } from '@openmrs/esm-framework';
+import myImage from '/home/user/Documents/GitHub/openmrs-esm-core/packages/apps/esm-login-app/src/public/ohri_logo_light.svg';
 
 export const configSchema = {
+  appVersion: {
+    _type: Type.String,
+    _default: '2.0',
```
Comment (pirupius) [inline packages/apps/esm-login-app/src/config-schema.ts:7]: 3.x

File: packages/apps/esm-login-app/src/config-schema.ts [hunk 4]
```diff
@@ -1,6 +1,22 @@
...
+    _description: "Specifies the version of the configuration. Set to '2.0' for this configuration.",
```
Comment (pirupius) [inline packages/apps/esm-login-app/src/config-schema.ts:8]: don't include the number in the description

File: packages/apps/esm-login-app/src/config-schema.ts [hunk 5]
```diff
@@ -1,6 +1,22 @@
...
+  },
+  showVersionNumber: {
+    _type: Type.Boolean,
+    _default: true,
+    _description: 'some random words about what this d',
```
Comment (pirupius) [inline packages/apps/esm-login-app/src/config-schema.ts:13]: U were supposed to update this

File: packages/apps/esm-login-app/src/config-schema.ts [hunk 6]
```diff
@@ -1,6 +1,22 @@
...
+  },
+  partnerLogos: {
+    _type: Type.Array,
+    _default: [myImage],
+    _description: 'A path or URL to an image. If null, will use the OHRI SVG sprite.',
```
Comment (pirupius) [inline packages/apps/esm-login-app/src/config-schema.ts:18]: This changes have to go to core which is being used by the whole community so it shouldn't have implementation specific names or descriptions.

File: packages/apps/esm-login-app/src/config-schema.ts [hunk 7]
```diff
@@ -82,6 +98,21 @@ export interface ConfigSchema {
...
+  showVersionNumber: {
+    type: boolean;
+    _default: boolean;
+    _description: string;
+  };
+  partnerLogo: {
+    type: string;
+    _default: any[];
+    _description: string;
+  };
```
Comment (pirupius) [inline packages/apps/esm-login-app/src/config-schema.ts:115]: These are repeated, please delete

File: packages/apps/esm-login-app/src/declarations.d.ts [hunk 8]
```diff
@@ -4,5 +4,6 @@ declare type Observable<T> = import('rxjs').Observable<T>;
 
 declare module '*.css';
 declare module '*.scss';
+declare module '*.svg';
```
Comment (pirupius) [inline packages/apps/esm-login-app/src/declarations.d.ts:7]: revert this

File: packages/apps/esm-login-app/src/login/login.component.tsx [hunk 9]
```diff
@@ -269,12 +272,20 @@ const Login: React.FC<LoginProps> = () => {
...
-            <svg role="img" className={styles['powered-by-logo']}>
-              <use xlinkHref="#omrs-logo-partial-mono"></use>
-            </svg>
+            <div className={styles['myLogos']}>
+              <div className={styles.omrsLogo}>
+                <svg role="img" className={styles['powered-by-logo']}>
+                  <use xlinkHref="#omrs-logo-partial-mono"></use>
+                </svg>
+              </div>
+              <div className={styles.ohriLogo}>{partnerLogoImgs}</div>
```
Comment (pirupius) [inline packages/apps/esm-login-app/src/login/login.component.tsx:287]: Add a condition to only render if there partner logos

File: packages/apps/esm-login-app/src/public/ohri_logo_light.svg
Comment (pirupius) [inline packages/apps/esm-login-app/src/public/ohri_logo_light.svg:1]: Delete this. This is meant to be configured at the implementation level i.e within the config file in the distro

File: packages/framework/esm-framework/docs/API.md [hunk 10]
```diff
@@ -210,7 +210,7 @@
 
 #### Defined in
 
-[packages/framework/esm-api/src/shared-api-objects/current-patient.ts:6](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-api/src/shared-api-objects/current-patient.ts#L6)
+[packages/framework/esm-api/src/shared-api-objects/current-patient.ts:6](https://github.com/kirwea/openmrs-esm-core/blob/main/packages/framework/esm-api/src/shared-api-objects/current-patient.ts#L6)
```
Comment (pirupius) [inline packages/framework/esm-framework/docs/API.md:213]: Revert all changes in this file and the other unnecessary changes

**Review by pirupius — COMMENTED** [review]
The only changes that needed to be made on this branch are the ones related to the task. So that doesn't include updating the `.md` files

File: package.json (hunk 1 above)
Comment (kirwea) [reply package.json:12]: hero said this was the correct url to access and fetch from the backend

File: packages/apps/esm-login-app/src/declarations.d.ts (hunk 8 above)
Comment (kirwea) [reply packages/apps/esm-login-app/src/declarations.d.ts:7]: ok,and what about the error that will come when i do?the ohri logo needs the that declaration to display

File: packages/apps/esm-login-app/src/public/ohri_logo_light.svg
Comment (kirwea) [reply packages/apps/esm-login-app/src/public/ohri_logo_light.svg:1]: pius honestly now i am in the dark,uuuum....i created that file to use an an svg for the ohri logo to display

File: package.json (hunk 1 above)
Comment (pirupius) [reply package.json:12]: Please remember our discussion about this. And read the other comments about this being a community repo
